arbos/arbostypes/incomingmessage.go

```diff
@@ -127,14 +127,21 @@ func (msg *L1IncomingMessage) Equals(other *L1IncomingMessage) bool {
 	return msg.Header.Equals(other.Header) && bytes.Equal(msg.L2msg, other.L2msg)
 }
 
+func hashesEqual(ha, hb *common.Hash) bool {
+	if (ha == nil) != (hb == nil) {
+		return false
+	}
+	return (ha == nil) && (hb == nil) || *ha == *hb
```

Comment:
Same nit here about explicit parenthesis around the `&&`

Reply:
Done.